py4DSTEM/process/diffraction/crystal_phase.py

```diff
@@ -33,306 +35,1366 @@ def __init__(
+        scale_markers_calculated: float
+            Size of the calculate diffraction peak markers.
+        crystal_inds_plot: tuple of ints
+            Which crystal index / indices to plot.
+        phase_colors: np.array
+            Color of each phase, should have shape = (num_phases, 3)
+        figsize: (float,float)
+            Size of the output figure.
+        verbose: bool
+            Print the resulting fit weights to console.
+        returnfig: bool
+            Return the figure and axis handles for the plot.
+
+
+        Returns
+        -------
+        phase_weights: (np.array)
+            Estimated relative fraction of each phase for all probe positions.
+            shape = (num_x, num_y, num_orientations)
+            where num_orientations is the total number of all orientations for all phases.
+        phase_residual: (np.array)
+            Residual intensity not represented by the best fit phase weighting for all probe positions.
+            shape = (num_x, num_y)
+        phase_reliability: (np.array)
+            Estimated reliability of match(es) for all probe positions.
+            Typically calculated as the best fit score minus the second best fit.
+            shape = (num_x, num_y)
+        int_total: (np.array)
+            Sum of experimental peak intensities for all probe positions.
+            shape = (num_x, num_y)
+        fig,ax: (optional)
+            matplotlib figure and axis handles
 
-        Args:
-            map_scale_values (float):   Value to scale correlations by
         """
-        phase_maps = []
-        if map_scale_values is None:
-            map_scale_values = [1] * len(self.orientation_maps)
-        corr_sum = np.sum(
-            [
-                (self.orientation_maps[m].corr[:, :, index] * map_scale_values[m])
-                for m in range(len(self.orientation_maps))
-            ]
-        )
-        for m in range(len(self.orientation_maps)):
-            phase_maps.append(self.orientation_maps[m].corr[:, :, index] / corr_sum)
-        show_image_grid(lambda i: phase_maps[i], 1, len(phase_maps), cmap="inferno")
-        return
-
-    def plot_phase_map(self, index=0, cmap=None):
-        corr_array = np.dstack(
-            [maps.corr[:, :, index] for maps in self.orientation_maps]
-        )
-        best_corr_score = np.max(corr_array, axis=2)
-        best_match_phase = [
-            np.where(corr_array[:, :, p] == best_corr_score, True, False)
-            for p in range(len(self.orientation_maps))
-        ]
-
-        if cmap is None:
-            cm = plt.get_cmap("rainbow")
-            cmap = [
-                cm(1.0 * i / len(self.orientation_maps))
-                for i in range(len(self.orientation_maps))
-            ]
 
-        fig, (ax) = plt.subplots(figsize=(6, 6))
-        ax.matshow(
-            np.zeros((self.orientation_maps[0].num_x, self.orientation_maps[0].num_y)),
-            cmap="gray",
+        # tolerance
+        tol2 = 1e-6
```

Comment:
This could have a better name

Reply:
updated to be tolerance_origin_2, explanatory comment added too.